### PR TITLE
Add TARGWP20, TARGWP100 and TARGWP500 from third assessment report

### PR DESCRIPTION
**Pull request by mikapfl**
We actually found a usage of global warming potentials from the third assessment report in the wild; therefore, we need them. Source is linked. I transferred it from pdf by hand (a mind-numbing exercise), so there might be errors. I'm willing to re-check tomorrow, but today it wouldn't make much sense.

### CODE REVIEW

**Comment by znicholls** [comment]
I'll take a look now @mikapfl

File: globalwarmingpotentials.csv [hunk 1]
```diff
@@ -1,98 +1,115 @@
...
-HFE374pc2,,,627,758,12.5,45,3.56,2.25
-PFPMIE,,10300,9710,10789,10300,7750,11700,12000
-CHCl3,4,,16,20,20.6,74.2,5.87,3.72
-CH2Cl2,9,8.7,9,11,11.2,40.2,3.18,2.01
-CH3Cl,,13,12,15,5.54,19.9,1.58,1
-Halon1201,,,376,454,380,1340,108,69.8
+# Global warming potentials of greenhouse gases.,,,,,,,,,,,
+# Sources:,,,,,,,,,,,
+#   - SARGWP100: https://ghgprotocol.org/calculation-tools (specifically https://www.ghgprotocol.org/sites/default/files/ghgp/Global-Warming-Potential-Values%20%28Feb%2016%202016%29_1.pdf),,,,,,,,,,,
+#   - TARGWP100, TARGWP20, TARGWP500: https://www.ipcc.ch/site/assets/uploads/2018/03/WGI_TAR_full_report.pdf, page 388 Table 6.7,,,,,,,,,,,
```
Comment (znicholls) [inline globalwarmingpotentials.csv]:
```suggestion
#   - TARGWP100, TARGWP20, TARGWP500: https://www.ipcc.ch/site/assets/uploads/2018/03/WGI_TAR_full_report.pdf, page 388 Table 6.7,,,,,,,,
```

To preserve GitHub's preview of the csv (this is stupid, but we appear to be trying to maintain this, see e.g. https://github.com/openclimatedata/globalwarmingpotentials/pull/27).

@rgieseke do we want to add an action to check that the number of rows is consistent?

File: globalwarmingpotentials.csv [hunk 2]
```diff
@@ -1,98 +1,115 @@
...
+CH4,21,23,25,28,34,27.9,62,81.2,7,7.95,5.38
+N2O,310,296,298,265,298,273,275,273,156,130,233
+CFC11,3800,4600,4750,4660,5352,6230,6300,8320,1600,2090,3540
+CFC12,8100,10600,10900,10200,11547,12500,10200,12700,5200,5710,10400
+CFC13,,14000,14400,13900,15451,16200,10000,12400,16300,17500,18800
+CFC113,4800,6000,6130,5820,6586,6520,6100,6860,2700,2830,5210
+CFC114,,9800,10000,8590,9615,9430,7500,8260,8700,6150,9410
+CFC115,,7200,7370,7670,8516,9600,4900,7410,9900,9880,11000
+Halon1301,5400,6900,7140,6290,7154,7200,7900,8320,2700,2750,5060
+Halon1211,,1300,1890,1750,2070,1930,36000,4920,390,552,406
```
Comment (znicholls) [inline globalwarmingpotentials.csv]:
```suggestion
Halon1211,,1300,1890,1750,2070,1930,3600,4920,390,552,406
```

File: globalwarmingpotentials.csv [hunk 3]
```diff
@@ -1,98 +1,115 @@
...
+SF5CF3,,,17700,17400,19396,18500,,13900,,21100,21600
+cC3F6,,,,9200,10208,,,,,,
+HFE125,,14900,14900,12400,13951,14300,12900,13500,9200,7680,13100
+HFE134,,6100,6320,5560,6512,6630,10500,12700,2000,1940,2060
+HFE143a,,750,756,523,632,616,2500,2170,230,176,113
+HCFE235da2,,340,350,491,595,539,1100,1930,110,154,98.4
+HFE245cb2,,580,708,654,790,747,1900,2630,180,213,137
+HFE245fa2,,570,659,812,981,878,1900,3060,180,251,162
+HFE254cb2,,30,,,,,99,,9,,
+HFE347mcc3,,,575,530,641,576,1600,2020,150,164,106
```
Comment (znicholls) [inline globalwarmingpotentials.csv]:
```suggestion
HFE347mcc3,,480,575,530,641,576,1600,2020,150,164,106
```

I think there's a value that can be used, or did I miss something?

File: globalwarmingpotentials.csv [hunk 4]
```diff
@@ -1,98 +1,115 @@
...
+HFE347pcf2,,,580,889,1072,980,,3370,,279,181
+HFE356pcc3,,110,110,413,500,277,,995,,79,50.4
```
Comment (znicholls) [inline globalwarmingpotentials.csv]:
```suggestion
HFE356pcf3,,110,110,413,500,277,,995,,79,50.4
```

Hmm, in TAR it is HFE356pcf3. Previously we had HFE356pcc3. I have no idea if these are meant to be the same and one is just a typo, or whether they're actually different things...

File: globalwarmingpotentials.csv [hunk 5]
```diff
@@ -1,98 +1,115 @@
...
+HFE236ea2,,960,,1790,2143,2590,3100,7020,300,741,521
+HFE236fa,,470,,979,1177,1100,1600,3670,150,315,205
+HFE245fa1,,280,,828,997,934,940,3170,86,266,173
+HFE263fb2,,11,,1,2,,37,,3,,
+HFE329mcc2,,890,,3070,3598,3770,2800,7550,280,1100,1090
+HFE338mcf2,,540,,929,1118,1040,1800,3460,170,297,194
+HFE347mcf2,,360,,854,1028,963,1200,3270,110,275,179
+HFE356mec3,,98,,387,468,264,330,949,30,75.3,48
+HFE356pcf2,,260,,719,867,831,860,2870,80,237,154
+HFE356pcf3,,430,,446,540,484,1500,1730,130,138,88.4
```
Comment (znicholls) [inline globalwarmingpotentials.csv:95]: TAR has HFE356pcc3, other reports have HFE356pcf3. Not sure if one is a typo or whether they're actually different.

**Review by znicholls — CHANGES_REQUESTED** [review]
Nice. A few typos and questions I think. Was indeed hard going through this with a pdf.

Is there a reason we're adding extra gases like "HGalden1040x"? Or is it just cause we've read them now, so may as well keep them?

File: globalwarmingpotentials.csv (hunk 1 above)
Comment (rgieseke) [reply globalwarmingpotentials.csv]: @znicholls Ye, that could be useful. Though, if one wants to keep the GitHub rendering used, actual readability might be improved by moving additional commas to the very end of the commented '#' lines and use ';' or something else to separate 'TARGWP100' and 'TARGWP20'.

File: globalwarmingpotentials.csv (hunk 5 above)
Comment (rgieseke) [reply globalwarmingpotentials.csv:95]: This document https://faolex.fao.org/docs/pdf/mex148237.pdf has

```
HFE356pcc3; 1,1,2,2,3,3-Hexafluoropropil metil éter; (CH3OCF2CF2CHF2): CAS 160620202
```

and
```
HFE356pcf3; 2,2,3,3-Tetrafluoropropil difluorometil éter; (CHF2OCH2CF2CHF2): CAS
35042990
```

In general my feeling would be to include the data as published in the original IPCC sources, unless it's an obvious typo.

File: globalwarmingpotentials.csv (hunk 4 above)
Comment (mikapfl) [reply globalwarmingpotentials.csv]: I think you are a bit confused here, both exist and we have values for both:

```csv
HFE356pcc3,,110,110,413,500,277,,995,,79,50.4
HFE356pcf3,,430,,446,540,484,1500,1730,130,138,88.4
```

File: globalwarmingpotentials.csv (hunk 5 above)
Comment (mikapfl) [reply globalwarmingpotentials.csv:95]: As I said above, both exist, we have values for both, they are distinct.

Comment (rgieseke) [reply globalwarmingpotentials.csv:95]: Even better, @mikapfl :-)

**Comment by mikapfl** [comment]
Note: I didn't include `-(CF2)4CH(OH)-` because the `-` are significant and we can't deal with them easily and it's just not worth the headache to figure something out given this is likely not used anywhere we care about.

**Comment by rgieseke** [comment]
> Note: I didn't include -(CF2)4CH(OH)- because the - are significant and we can't deal with them easily and it's just not worth the headache to figure something out given this is likely not used anywhere we care about.

Where is this not working? Should be a valid JS object or Python dictionary key, isn't it?

**Comment by mikapfl** [comment]
> Where is this not working? Should be a valid JS object or Python dictionary key, isn't it?

It won't work as a normal unit in pint, i.e. in openscm-units, I think.

**Comment by mikapfl** [comment]
Thanks for checking everything, Zeb! Very much appreciated. To be extra sure, I now did an alternative extraction using https://smallpdf.com/pdf-to-excel and targeted copy+pasting, which means I didn't have the chance to introduce typos using this way. I found two missing values for HFE356pcc3 (I was apparently also confused, it is super confusing because the gases are not really ordered alphabetically or something).

Now I'm very confident that the values included here reflect what's in the official pdf.

**Comment by mikapfl** [comment]
> Is there a reason we're adding extra gases like "HGalden1040x"? Or is it just cause we've read them now, so may as well keep them?

I just figured, if it is the pdf, let's add it now while we deal with this pdf and then never have to think about it again, hopefully.

**Comment by rgieseke** [comment]
> >   Where is this not working? Should be a valid JS object or Python dictionary key, isn't it?
>
> It won't work as a normal unit in pint, i.e. in openscm-units, I think.

Maybe openscm-units should then just ignore it (it's very likely not handled by any SCM I guess). Not sure what level of completeness we have for other reports. If we include 'HGalden1040x' then why not '-(CF2)4CH(OH)-'?

**Comment by mikapfl** [comment]
> Maybe openscm-units should then just ignore it (it's very likely not handled by any SCM I guess). 

Ok, I added it and will deal with the fallout in openscm-units.

I did the same amount of checking for it (extract via two different routes) so am confident in the values as well.

**Comment by znicholls** [comment]
> Now I'm very confident that the values included here reflect what's in the official pdf.

Nice!

> Ok, I added it and will deal with the fallout in openscm-units

Hmph, but fair I guess

File: README.md [hunk 6]
```diff
@@ -15,6 +15,8 @@ Global warming potentials of greenhouse gases.
 
 - Second Assessment Report (SAR) - **SARGWP100**
   [Data source](https://www.ghgprotocol.org/sites/default/files/ghgp/Global-Warming-Potential-Values%20%28Feb%2016%202016%29_1.pdf)
+- Third Assessment Report (TAR) - **TARGWP100**
+  [Data source](https://www.ipcc.ch/site/assets/uploads/2018/03/WGI_TAR_full_report.pdf), page 388 Table 6.7
```
Comment (znicholls) [inline README.md:19]: @mikapfl this reference is different to the one below (which references the synthesis report), however this reference seemed to have all the data. Am I missing something or should we just make them all the same reference?

**Review by znicholls — APPROVED** [review]
Super, thank you!

File: README.md (hunk 6 above)
Comment (mikapfl) [reply README.md:19]: Argh, I forgot to update the reference everywhere. The full report contains more species than the synthesis report. I started extraction with the synthesis report, then switched to the full report at some point.

**Comment by rgieseke** [comment]
Released as 0.10.0, thanks @mikapfl and @znicholls!

**Comment by rgieseke** [comment]
0.10.1 - forgot to re-run `make`.

**Comment by mikapfl** [comment]
> 0.10.1 - forgot to re-run make.

Something which could be included in pre-commit or other CI :wink:

**Pull request merged**